src/services/expenses.service.js

```diff
@@ -0,0 +1,76 @@
+let expenses = [];
+let nextId = 1;
+
+const getAllExpenses = () => {
+  return expenses;
+};
+
+const getById = (id) => {
+  return expenses.find((exp) => exp.id === parseInt(id));
+};
+
+const normalizeCategory = (categories) => {
+  return Array.isArray(categories) || !categories ? categories : [categories];
+};
+
+const filterExpensesById = (id) => {
+  expenses = expenses.filter((exp) => exp.userId === parseInt(id));
+
+  return expenses;
+};
+
+const filterExpensesByCategory = (category) => {
+  return expenses.filter((exp) => category.includes(exp.category));
+};
+
+const filterExpensesByDate = (from, to) => {
+  return expenses.filter((exp) => {
+    const spentAt = new Date(exp.spentAt);
+
+    return spentAt >= from && spentAt <= to;
+  });
+};
+
+const create = ({ userId, spentAt, title, amount, category, note }) => {
```

Comment:
You are destructuring the properties from the argument directly in the function parameters. While this is not necessarily a bad practice, it can make the function signature less clear. Consider destructuring within the function body for better readability.